app/scripts/modules/core/cluster/task.matcher.ts

```diff
@@ -0,0 +1,92 @@
+import {has, intersection, map} from 'lodash';
+
+import {ITask} from '../task/task.read.service';
+import {ServerGroup} from '../domain/serverGroup';
+
+export interface ITaskMatcher {
+ (task: ITask, serverGroup: ServerGroup): boolean;
+}
+
+function createcopylastasgMatcher(task: ITask, serverGroup: ServerGroup): boolean {
```

Comment:
The matcher names are keyed off the `notification.type` field in the task context, which is all lower case. No idea why we decided to make them all lower case, but we did. Newer ones, e.g. `rollbackServerGroup`, is camel cased. Go figure.

The `function`s in this file could be camel cased, but I decided to leave them in the same case as the value they're matching, just to be sort of consistent.